lib/id3c/cli/command/de-identify.py

```diff
@@ -0,0 +1,89 @@
+"""
+De-identify personally identifiable information (PII) by generating a hash.
+"""
+import click
+import logging
+import os
+import hashlib
+import pandas as pd
+from sys import stdout
+from typing import Tuple
+from id3c.cli import cli
+from id3c.cli.command import load_input_from_file_or_stdin
+
+
+LOG = logging.getLogger(__name__)
+SUPPORTED_EXTENSIONS = ('.csv', '.tsv', '.xls', '.xlsx')
+
+@cli.command("de-identify")
+@click.argument("columns",
+    metavar = "<columns>",
+    nargs = -1)
+@click.option("--filename",
+    metavar = "<filename.{csv,tsv,xls,xlsx}>",
+    type = click.Path(exists=True))
+@click.option("--drop-input-columns",
+    is_flag = True,
+    help = "Optional flag to drop input columns from output")
+
+def de_identify(columns, drop_input_columns, filename=None):
+    """
+    Given a <filename.{csv,tsv,xls,xlsx}> that contains personally identifiable
+    information (PII), generate a hash using specified <columns> and an environment
+    variable, "PARTICIPANT_DEIDENTIFIER_SECRET".
+
+    <columns> are column names of fields to include in the hash. All values are
+    joined in the order of of the column names, so keep the order of column
+    names consistent!
+
+    <columns> values are expected to be canonicalized before running this
+    command!
+
+    If no <filename.{csv,tsv,xls,xlsx} is provided, then try to read data from
+    stdin, assuming data formatted as comma-separated values. This is expected when
+    piping output from `id3c geocode` directly into this command.
+
+    Generated hashes are output to stdout along with input data as
+    comma-separated values. Original input can be dropped with the
+    --drop-input-columns flag.
+    """
+    input_df = load_input_from_file_or_stdin(filename, SUPPORTED_EXTENSIONS)
+    fields_to_include = extract_fields_from_input(input_df, columns)
+    joined_fields = fields_to_include.apply(lambda x: ' '.join(x), axis=1)
+
+    hash_secret = os.environ['PARTICIPANT_DEIDENTIFIER_SECRET']
+    hashes = joined_fields.map(lambda x: generate_hash(x, hash_secret) if x else None)
+
+    if drop_input_columns:
+        output_df = pd.DataFrame(hashes, columns = ['hash'])
+    else:
+        output_df = input_df
+        output_df['hash'] = hashes
```

Comment:
Similar comment as before: 
I think I would prefer to build the output dataframes the same way, dropping or appending the desired columns after construction.

Reply:
Agreed. The current implementation of `--drop-input-columns` means that all columns are dropped, not just those specified as arguments / input to the hash.